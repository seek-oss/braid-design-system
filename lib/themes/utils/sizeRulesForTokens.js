import merge from 'lodash/merge';
import createResponsiveRules from './createResponsiveRules';

export default ({ tokens, property }) => {
  const rules = [];

  Object.keys(tokens.text).forEach(typeName => {
    const type = tokens.text[typeName];

    rules.push(
      createResponsiveRules({
        tokens,
        selector: `.${typeName}Text`,
        mobileRules: {
          [property]: `${type.mobile.rows * tokens.rowHeight}px`
        },
        desktopRules: {
          [property]: `${type.desktop.rows * tokens.rowHeight}px`
        }
      })
    );

    rules.push(
      createResponsiveRules({
        tokens,
        selector: `.${typeName}TextInline`,
        mobileRules: {
          [property]: `${type.mobile.size}px`
        },
        desktopRules: {
          [property]: `${type.desktop.size}px`
        }
      })
    );
  });

  return merge({}, ...rules);
};
