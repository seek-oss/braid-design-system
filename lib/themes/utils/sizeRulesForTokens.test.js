import sizeRulesForTokens from './sizeRulesForTokens';
import forEachTheme from '../../test/utils/forEachTheme';

describe('sizeRulesForTokens', () => {
  forEachTheme(({ theme }) => {
    test('Rules', () => {
      expect(
        sizeRulesForTokens({ tokens: theme.tokens, property: 'width' })
      ).toMatchSnapshot();
    });
  });
});
