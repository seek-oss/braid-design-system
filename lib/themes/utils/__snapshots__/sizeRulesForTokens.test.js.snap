// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sizeRulesForTokens Theme: jobStreet Rules 1`] = `
Object {
  ".largeText": Object {
    "width": "24px",
  },
  ".largeTextInline": Object {
    "width": "18px",
  },
  ".standardText": Object {
    "width": "20px",
  },
  ".standardTextInline": Object {
    "width": "14px",
  },
  "@media screen and (min-width: 768px)": Object {
    ".largeText": Object {
      "width": "28px",
    },
    ".largeTextInline": Object {
      "width": "20px",
    },
  },
}
`;

exports[`sizeRulesForTokens Theme: seekAnz Rules 1`] = `
Object {
  ".largeText": Object {
    "width": "24px",
  },
  ".largeTextInline": Object {
    "width": "18px",
  },
  ".standardText": Object {
    "width": "24px",
  },
  ".standardTextInline": Object {
    "width": "16px",
  },
}
`;

exports[`sizeRulesForTokens Theme: seekAsia Rules 1`] = `
Object {
  ".largeText": Object {
    "width": "24px",
  },
  ".largeTextInline": Object {
    "width": "18px",
  },
  ".standardText": Object {
    "width": "24px",
  },
  ".standardTextInline": Object {
    "width": "16px",
  },
  "@media screen and (min-width: 768px)": Object {
    ".largeText": Object {
      "width": "28px",
    },
    ".largeTextInline": Object {
      "width": "20px",
    },
  },
}
`;

exports[`sizeRulesForTokens Theme: wireframe Rules 1`] = `
Object {
  ".largeText": Object {
    "width": "24px",
  },
  ".largeTextInline": Object {
    "width": "18px",
  },
  ".standardText": Object {
    "width": "24px",
  },
  ".standardTextInline": Object {
    "width": "16px",
  },
}
`;
